components/chapters/LoginModal.tsx

```diff
@@ -0,0 +1,196 @@
+      contentLabel="Login Modal"
+    >
+      <div className="float-right flex justify-end">
+        <button onClick={props.onClose} aria-label="Close">
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            className="h-6 w-6"
+            fill="none"
+            viewBox="0 0 24 24"
+            stroke="currentColor"
+          >
+            <path
+              strokeLinecap="round"
+              strokeLinejoin="round"
+              strokeWidth={2}
+              d="M6 18L18 6M6 6l12 12"
+            />
+          </svg>
+        </button>
+      </div>
+
+      <div className="p-4">
+      {!loaded && !props.signedIn && (
+        <>
+          <img src="/assets/avatars/1.png" width={80} height={80} />
+          <h2 className="mb-4 mt-5 text-3xl font-bold">Load your progress</h2>
+          <p className="mb-5 text-xl">
+            Enter your personal code below to restore your chapter and challenge progress, as well as your avatar. <a href="#" className='text-white underline'>No code yet?</a>
+          </p>
+
+          <div className="flex flex-col justify-between">
+            <div className="flex w-full">
+              <input
+                className="bg-transparent outline-none text-white border-dotted border-white border-2 p-1 w-full text-lg"
+                type="text"
+                placeholder="Enter your code"
+                value={userPrivateKey}
+                onChange={(e) => setPrivateKey(e.target.value)}
+              />
+            </div>
+            <div className="flex w-full mt-4">
+              <button disabled={!userPrivateKey} onClick={loadProgress} className={`border-white border-2 w-full p-1 text-xl ${!userPrivateKey && 'opacity-50'}`}>Load my progress</button>
+            </div>
+            <div className="flex w-full mt-4 mb-4">
+              {userPrivateKey && !checkValidKey() && (
+                <div className='flex items-center'>
+                  <svg width="30" height="30" viewBox="0 0 30 30" fill="none" xmlns="http://www.w3.org/2000/svg">
+                    <rect width="30" height="30" rx="15" fill="#F3241D" fill-opacity="0.25"/>
+                    <circle cx="15" cy="10.25" r="1.25" fill="#F3241D"/>
+                    <rect x="14" y="13" width="2" height="8" fill="#F3241D"/>
+                  </svg>
+                  <p className="text-red-500 text-sm ml-2">Sorry, that’s not the right code for the progress stored in this browser.</p>
+                </div>
+              )}
+            </div>
+
+            <div className='flex items-center mt-auto'>
+                <p>
+                  Want to start again? <button onClick={clearProgress} className='underline'>Clear saved progress</button>.
+                </p>
+            </div>
+          </div>
+        </>)}
+      {props.signedIn && (
+        <>
+          <Image
```

Comment:
Is it worth putting creating an avatar component? We'll show it quite often in the mining lesson in a few sizes and could bake those options in.

Reply:
Done in 3d0cd738f917ab419773aee1fd932256d118f13c.